### PR TITLE
[GDScript] Fix `get_method` for lambda self `Callable`s

### LINKED ISSUE #94074: `Callable.get_method()` does not fully work on lambda functions if they access member variables
### Tested versions

v4.2.2.stable.official [15073afe3]

### System information

Godot v4.2.2.stable - Windows 10.0.22631 - Vulkan (Forward+) - integrated Intel(R) UHD Graphics (Intel Corporation; 27.20.100.9415) - Intel(R) Core(TM) i5-1035G1 CPU @ 1.00GHz (8 Threads)

### Issue description

```gdscript
extends Node

var member := 3

func _ready() -> void:
    var callable1 := func test() -> int:
        return member
    var callable2 := func test() -> int:
        return 0

    print(callable1.get_method()) # error, prints nothing
    print(callable2.get_method()) # no error, prints 'test'
```
For lambda functions that access variables from the object that the function was defined on, `Callable.get_method()` does not work. Instead, an error is printed and an empty `StringName` is returned.
```
E 0:00:01:0170   Test.gd:11 @ _ready(): Can't get method on CallableCustom "test(lambda)".
  <C++ Error>    Method/function failed. Returning: StringName()
  <C++ Source>   core/variant/callable.cpp:393 @ get_method()
  <Stack Trace>  Test.gd:11 @ _ready()
```
There is already a closed issue about a very similar bug (#80434), but the solution did not fix `get_method()` for these lambda functions, only when they don't access member variables.

As a workaround, you can declare a static variable and access the members from there:
```gdscript
var member := 3
static var this: Node

func _ready() -> void:
    this = self

    var callable1 := func test() -> int:
        return this.member
    var callable2 := func test() -> int:
        return 0

    print(callable1.get_method()) # no error anymore, prints 'test'
    print(callable2.get_method()) # no error, prints 'test'
```
I did a bit of digging and it seems like there was/is big code duplication between 2 kinds of lambda functions: 1 that does not access member variables and 1 that does. So it seems like the fix from #80434 may have only changed 1 and not the other. Though I'm no expert when it comes to C++ and the godot source code.

### Steps to reproduce

See above

### Minimal reproduction project (MRP)

N/A

### PR DESCRIPTION
**Pull request by AThousandShips**
* Fixes: https://github.com/godotengine/godot/issues/94074

### CODE REVIEW

**Review by adamscott — APPROVED** [review]
As you said, it was a simple oversight.

**Comment by dalexeev** [comment]
It would be nice to add a test.

**Comment by AThousandShips** [comment]
I'll make one

File: modules/gdscript/tests/scripts/runtime/features/lambda_get_method.gd
Comment (dalexeev) [inline modules/gdscript/tests/scripts/runtime/features/lambda_get_method.gd:1]: Also anonymous lambdas should have the name `<anonymous lambda>`:

https://github.com/godotengine/godot/blob/ec02d406ca0b9c822addff49cf58e9a72cf74eb0/doc/classes/Callable.xml#L168-L173

Comment (AThousandShips) [reply modules/gdscript/tests/scripts/runtime/features/lambda_get_method.gd:1]: I'll add a test for that, but this just copies the code from `GDScriptLambdaCallable` so if that doesn't return the correct case then I think it'd be best to fix it in a dedicated PR as this just fixes the inconsistency

Comment (AThousandShips) [reply modules/gdscript/tests/scripts/runtime/features/lambda_get_method.gd:1]: At least the code for `get_as_text` returns `(anonymous lambda)` in that case, so I think that should be a separate PR fixing that up there too

Comment (AThousandShips) [reply modules/gdscript/tests/scripts/runtime/features/lambda_get_method.gd:1]: It does work correctly, so adding test for it, no need for further change

**Review by dalexeev — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Comment by AThousandShips** [comment]
Thank *you*!

**Pull request merged**